### PR TITLE
Add new runner command for running fuzz targets

### PR DESCRIPTION
**Pull request by rbehjati**
Ref #2047

### CODE REVIEW

File: runner/src/internal.rs [hunk 1]
```diff
@@ -293,6 +295,41 @@ impl RustBinaryOptions for BuildServer {
...
+    #[serde(default)]
+    pub bin: Vec<CargoBinary>,
+}
+
+/// Partial information about a Cargo binary, as included in a Cargo manifest.
+#[derive(serde::Deserialize, Debug)]
+pub struct CargoBinary {
+    #[serde(default)]
+    pub name: String,
+}
```
Comment (tiziano88) [inline runner/src/internal.rs:331]: Would https://crates.io/crates/cargo_metadata work here? (we already use it elsewhere: https://github.com/project-oak/oak/blob/9f0ac739e47c6b713b3fd2838ad0ce4738192761/sdk/rust/oak_tests/src/lib.rs#L50-L53)

File: runner/src/main.rs [hunk 2]
```diff
@@ -176,6 +177,67 @@ fn run_tests_tsan() -> Step {
...
+    }
+}
+
+pub fn run_fuzz_targets_in_crate(
+    path: &PathBuf,
+    cargo_manifest: &CargoManifest,
+    opt: &RunFuzzTargets,
+) -> Step {
+    // `cargo-fuzz` can only run in the crate that contains the `fuzz` crate. So we need to use
+    // `Cmd::new_in_dir` to execute the command inside the crate's directory.
```
Comment (tiziano88) [inline runner/src/main.rs:215]: But why is the code popping two elements from the path?

**Review by tiziano88 — APPROVED** [review]

File: runner/src/internal.rs (hunk 1 above)
Comment (rbehjati) [reply runner/src/internal.rs:331]: This looks interesting, but I could not find any structs in `cargo_metadata` that matches the structure of a Cargo toml file. I particularly tried `Metadata`, and `Package`.

File: runner/src/main.rs (hunk 2 above)
Comment (rbehjati) [reply runner/src/main.rs]: Added an explanation.

**Pull request merged**